internal/controllers/gateway/backendtlspolicy_utils.go

```diff
@@ -0,0 +1,160 @@
+func (r *BackendTLSPolicyReconciler) getBackendTLSPolicyAncestors(ctx context.Context, policy gatewayapi.BackendTLSPolicy) ([]gatewayapi.Gateway, error) {
+	gateways := []gatewayapi.Gateway{}
+	for _, targetRef := range policy.Spec.TargetRefs {
+		if (targetRef.Group != "core" && targetRef.Group != "") && targetRef.Kind != "Service" {
+			continue
+		}
+
+		httpRoutes := gatewayapi.HTTPRouteList{}
+		if err := r.Client.List(ctx, &httpRoutes,
+			client.MatchingFields{httpRouteBackendRefIndexKey: policy.Namespace + "/" + string(targetRef.Name)},
+		); err != nil {
+			return nil, err
+		}
+
+		for _, httpRoute := range httpRoutes.Items {
+			for _, parentRef := range httpRoute.Spec.ParentRefs {
+				if *parentRef.Group != gatewayapi.V1Group || *parentRef.Kind != "Gateway" {
+					continue
+				}
+
+				namespace := httpRoute.Namespace
+				if parentRef.Namespace != nil {
+					namespace = string(*parentRef.Namespace)
+				}
+
+				var resolvedRefsStatus bool
+				// Check the resolvedRefs condition is set to true to ensure that all the references are properly resolved
+				// and granted by ReferenceGrants.
+				for _, parentStatus := range httpRoute.Status.Parents {
+					if parentStatus.ControllerName == GetControllerName() &&
+						*parentStatus.ParentRef.Group == *parentRef.Group &&
+						*parentStatus.ParentRef.Kind == *parentRef.Kind &&
+						parentStatus.ParentRef.Name == parentRef.Name &&
+						*parentStatus.ParentRef.Namespace == gatewayapi.Namespace(namespace) {
+						if _, found := lo.Find(parentStatus.Conditions, func(c metav1.Condition) bool {
+							return c.Type == string(gatewayapi.RouteConditionResolvedRefs) && c.Status == metav1.ConditionTrue
+						}); found {
+							resolvedRefsStatus = true
+							break
+						}
+					}
+				}
+				if resolvedRefsStatus {
+					gateway := &gatewayapi.Gateway{}
+					if err := r.Get(ctx, client.ObjectKey{Namespace: namespace, Name: string(parentRef.Name)}, gateway); err != nil {
+						// In case the object is not found, we don't want to return an error, but we want to continue.
+						if apierrors.IsNotFound(err) {
+							continue
+						}
+						return nil, err
+					}
+					gateways = append(gateways, *gateway)
+				}
+			}
+		}
+	}
+
+	return gateways, nil
+}
+
+// setPolicyStatus enforces an ancestorStatus for each Gateway associated to the given policy.
+// TODO: Conditions to the policy still to be implemented.
+func (r *BackendTLSPolicyReconciler) setPolicyStatus(ctx context.Context, policy gatewayapi.BackendTLSPolicy, gateways []gatewayapi.Gateway) error {
+	ancestors := []gatewayapi.PolicyAncestorStatus{}
+
+	// First copy all the ancestorstatuses managed by other controllers.
+	for _, ancestor := range policy.Status.Ancestors {
+		if ancestor.ControllerName == GetControllerName() {
+			continue
+		}
+		ancestors = append(ancestors, ancestor)
+	}
+
+	// Sort the Gateways to be consistent across subsequent reconcilation loops.
```

Comment:
```suggestion
	// Sort the Gateways to be consistent across subsequent reconciliation loops.
```
As linter suggested.